tests/consumer/test_oauth2.py

```diff
@@ -127,6 +129,24 @@ def test_authorized_url():
         )
 
 
+def test_authorized_url_invalid_response():
+    app, _ = make_app(debug=True)
+    with app.test_client() as client:
+        # reset the session before the request
+        with client.session_transaction() as sess:
+            sess["test-service_oauth_state"] = "random-string"
+        # make the request
+        with pytest.raises(MissingCodeError) as missingError:
+            client.get(
+                "/login/test-service/authorized?error_code=1349048&error_message=IMUSEFUL",
+                base_url="https://a.b.c",
+            )
+        assert str(missingError.value) == ('(missing_code) The redirect request did not contain a token. '
+                                           'Instead I got: {"error_message": "IMUSEFUL", "error_code": "1349048"}')
```

Comment:
When a dict is coerced to a string, the order of the key-value pairs is undefined. As a result, this test could fail on a different Python implementation that orders the key-value pairs differently. You could either do:
```python
assert str(missingError.value) in ('ordering 1', 'ordering 2')
```
or you can parse `missingError.value` back into a dict, and compare it with an actual dict.